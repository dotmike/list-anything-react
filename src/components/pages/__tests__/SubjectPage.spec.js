import React from "react";
import configureStore from "redux-mock-store";

import ConnectedSubjectPage, { UnconnectedSubjectPage } from "../SubjectPage";

const defaultProps = {
  history: {
    push: () => {}
  },
  match: {
    params: { _id: "1" }
  }
};

describe("ConnectedSubjectPage", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<ConnectedSubjectPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedSubjectPage", () => {
  const props = {
    ...defaultProps,
    fields: [],
    subjectDataArray: [],
    firstTab: {
      _id: "1"
    },
    fetchSubjectById: () => Promise.resolve(),
    fetchSDByTabId: () => Promise.resolve(),
    deleteSubject: () => Promise.resolve()
  };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<UnconnectedSubjectPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
