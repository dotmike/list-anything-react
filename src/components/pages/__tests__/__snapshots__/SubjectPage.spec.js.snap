// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSubjectPage renders correctly 1`] = `
<SubjectPage
  deleteSubject={[Function]}
  fetchSDByTabId={[Function]}
  fetchSubjectById={[Function]}
  fields={Array []}
  firstTab={
    Object {
      "_id": "",
    }
  }
  history={
    Object {
      "push": [Function],
    }
  }
  match={
    Object {
      "params": Object {
        "_id": "1",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  subjectDataArray={Array []}
/>
`;

exports[`UnconnectedSubjectPage renders correctly 1`] = `
<Segment
  basic={true}
  loading={true}
  style={
    Object {
      "height": "90%",
      "margin": "10px auto",
      "maxWidth": "90%",
    }
  }
>
  <div
    style={
      Object {
        "display": "inline-flex",
        "justifyContent": "space-between",
        "marginBottom": "3px",
        "width": "100%",
      }
    }
  >
    <ButtonGroup
      icon={true}
      size="medium"
    >
      <Button
        as="button"
        icon="sidebar"
        onClick={[Function]}
      />
      <Button
        as={[Function]}
        color="green"
        icon="plus"
        to="/subject/1/add"
      />
    </ButtonGroup>
    <EditDeleteBtnGroup
      onDelete={[Function]}
      onEdit={null}
      showDelete={true}
      showEdit={false}
    />
  </div>
</Segment>
`;
